net/module.c

```diff
@@ -63,7 +64,19 @@ static struct {
 	[M0_NET_XPRT_BULKMEM] = {
 		.name = "\"bulk-mem\" m0_net_xprt_module",
 		.xprt = &m0_net_bulk_mem_xprt
+	},
+#ifndef __KERNEL__
+	[M0_NET_XPRT_SOCK] = {
+		.name = "\"sock\" m0_net_xprt_module",
+		.xprt = (struct m0_net_xprt *)&m0_net_sock_xprt
```

Comment:
This is strange.
I know Nikita said, the `m0_net_sock_xprt ` should be defined as "const struct m0_net_xprt".
But all other transport are defined as "struct m0_net_xprt".

So, only this one needs a dynamic casting.

Reply:
Here, the "struct m0_net_xprt *xprt;" should be changed to "const struct m0_net_xprt *xprt;".